assets/src/edit-story/components/richText/karma/inlineStyleOverride.karma.js

```diff
@@ -0,0 +1,279 @@
+        expect(actual).toBe(expected);
+      });
+    });
+
+    it('should have correct formatting when already italic, then inline removing italic and adding bold, then inserting text', async () => {
+      // Enter edit-mode
+      await data.fixture.events.keyboard.press('Enter');
+
+      // Toggle italic for entire selection
+      await data.fixture.events.keyboard.shortcut('mod+i');
+
+      // Place cursor at start and then move to after second character
+      await setSelection(2, 2);
+
+      const {
+        italic,
+        bold,
+      } = data.fixture.editor.inspector.designPanel.textStyle;
+
+      // Verify that italic is toggled, bold is not
+      expect(italic.checked).toBe(true);
+      expect(bold.checked).toBe(false);
+
+      // (Un)toggle italic by button and then toggle bold by shortcut
+      // (also here the reverse would not work due to #1606 mentioned above)
+      await data.fixture.events.click(italic.button);
+      await data.fixture.events.keyboard.shortcut('mod+b');
+
+      // Verify that toggles are reversed
+      expect(italic.checked).toBe(false);
+      expect(bold.checked).toBe(true);
+
+      // Type "foo"
+      await data.fixture.events.keyboard.type('foo');
+
+      // Exit edit-mode
+      await data.fixture.events.keyboard.press('Escape');
+
+      // Expect correct result
+      const actual = getTextContent();
+      const expected =
+        '<span style="font-style: italic">Fi</span><span style="font-weight: 700">foo</span><span style="font-style: italic">ll in some text</span>';
+      expect(actual).toBe(expected);
+    });
+
+    it('should keep formatting when all text is replaced', async () => {
+      // Enter edit-mode
+      await data.fixture.events.keyboard.press('Enter');
+
+      // Make it all bold while selected
+      await data.fixture.events.keyboard.shortcut('mod+b');
+
+      // Replace text while entire text is selected
+      await data.fixture.events.keyboard.type('A new text');
+
+      // Exit edit-mode
+      await data.fixture.events.keyboard.press('Escape');
+
+      const actual = getTextContent();
+      const expected = '<span style="font-weight: 700">A new text</span>';
+      expect(actual).toBe(expected);
+    });
+
+    it('should keep formatting when all text is removed, then replaced', async () => {
+      // Enter edit-mode
+      await data.fixture.events.keyboard.press('Enter');
+
+      // Make it all bold while selected
+      await data.fixture.events.keyboard.shortcut('mod+b');
+
+      // Delete existing text by pressing backspace, then add new text
+      await data.fixture.events.keyboard.shortcut('Backspace');
+      await data.fixture.events.keyboard.type('Even more text');
+
+      // Exit edit-mode
+      await data.fixture.events.keyboard.press('Escape');
+
+      const actual = getTextContent();
+      const expected = '<span style="font-weight: 700">Even more text</span>';
+      expect(actual).toBe(expected);
```

Comment:
Maybe also add a test for when some part of the text has one font-weight, and then this part of the text is removed and then new text added to the same place.

E.g. 
With formatting: "Fil**l _in_ n**ew text", selection is on "in" -> Backspace -> type "Something". -- to verify the "Something" should have bold but no italic?

Reply:
Hm, makes sense. Will look into it.

Reply:
Added in 716485b72.